### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by JiangTianXiang**
Implement Redux for 
-Home
-Explore
-Search Photos
-Search Collections

### CODE REVIEW

File: src/pages/SearchCollectionResult/index.js [hunk 1]
```diff
@@ -16,133 +20,96 @@ import {
...
-  }
+  return (
+    <>
+      <LoadingBar color="#f11946" ref={ref} shadow={true} />
+      {hasData && (
+        <>
+          <PhotosAndSelectionsContainer>
+            <PhotoResultDetails>
+              <div>
+                Search results for "{searchTerm}"
```
Comment (yoloOnTheBattlefield) [inline src/pages/SearchCollectionResult/index.js:64]: this text doesn't do anything really?

File: src/pages/SearchCollectionResult/index.js [hunk 2]
```diff
@@ -16,133 +20,96 @@ import {
...
+              </StyledLink>
+              <UnderScoredLink
+                to={`/search/collections/${searchTerm}`}
+              >
+                Collections
+              </UnderScoredLink>
+            </PhotoSelectionSwitch>
+          </PhotosAndSelectionsContainer>
+          <InfiniteScroll
+            dataLength={renderObject[0].images.length}
```
Comment (yoloOnTheBattlefield) [inline src/pages/SearchCollectionResult/index.js]: nah, make a function that gets the length of the data, no hardcoded stuff please

File: src/pages/SearchPhotoResult/index.js [hunk 3]
```diff
@@ -16,129 +20,90 @@ import {
...
+              <UnderScoredLink to={`/search/photos/${searchTerm}`}>
+                Photos
+              </UnderScoredLink>
+              <StyledLink to={`/search/collections/${searchTerm}`}>
+                Collections
+              </StyledLink>
+            </PhotoSelectionSwitch>
+          </PhotosAndSelectionsContainer>
+          <InfiniteScroll
+            dataLength={renderObject[0].images.length}
```
Comment (yoloOnTheBattlefield) [inline src/pages/SearchPhotoResult/index.js]: 👆

File: src/store/searchCollection/searchCollectionReducer.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      return { ...state, page: state.page + 1 };
+    case SEARCH_FETCH_COLLECTION_SUCCESS:
+      return {
+        ...state,
+        data: [...state.data, ...action.payload.results],
+        renderObject: splitDataToColumns(
+          state.renderObject,
+          action.payload.results
+        ),
+        maxPage: action.payload.total_pages,
```
Comment (yoloOnTheBattlefield) [inline src/store/searchCollection/searchCollectionReducer.js:55]: python?

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/store/searchCollection/searchCollectionReducer.js (hunk 4 above)
Comment (JiangTianXiang) [reply src/store/searchCollection/searchCollectionReducer.js:55]: sorry not sure what do you mean here

File: src/pages/SearchCollectionResult/index.js (hunk 1 above)
Comment (JiangTianXiang) [reply src/pages/SearchCollectionResult/index.js:64]: This is just a div to show user the current search term. 
image

**Review by yoloOnTheBattlefield — APPROVED** [review]

**Pull request merged**